internal/testcmd/runner.go

```diff
@@ -0,0 +1,242 @@
+	// in the workflow configuration, if set.
+	projectDir string
+
+	// When enabled, outputs verbose output from the generator.
+	verboseOutput bool
+
+	// Discovered workflow configuration.
+	workflow *workflow.Workflow
+
+	// Name of the workflow target to run testing against. Defaults to all
+	// targets when an empty string.
+	workflowTarget string
+
+	// Enhanced CLI visualization tracker for the workflow.
+	workflowTracker *workflowTracking.WorkflowStep
+}
+
+// NewRunner creates a new Runner with the given options.
+func NewRunner(ctx context.Context, opts ...RunnerOpt) *Runner {
+	// Defaults
+	r := &Runner{
+		workflowTracker: workflowTracking.NewWorkflowStep("Workflow Testing", log.From(ctx), nil),
+	}
+
+	for _, opt := range opts {
+		opt(r)
+	}
+
+	return r
+}
+
+// Run loads the workflow, loads the generator, and runs target testing.
+func (r *Runner) Run(ctx context.Context) error {
+	if err := r.checkAccountType(ctx); err != nil {
+		return err
+	}
+
+	if err := r.loadWorkflow(); err != nil {
+		return err
+	}
+
+	if err := r.runWorkflowTargetTesting(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// RunWithVisualization calls Run with enhanced CLI visualizations. The full
+// logs are captured and only displayed if there is an error.
+func (r *Runner) RunWithVisualization(ctx context.Context) error {
+	logger := log.From(ctx)
+
+	// Swallow but retain the logs to be displayed later, upon failure
+	logCaptureBuffer := new(bytes.Buffer)
+	logCapture := logger.WithWriter(logCaptureBuffer)
+	updatesChannel := make(chan workflowTracking.UpdateMsg)
+
+	r.workflowTracker = workflowTracking.NewWorkflowStep("Workflow Testing", logCapture, updatesChannel)
+
+	var runErr error
+
+	runFnCli := func() error {
+		runCtx := log.With(ctx, logCapture)
+		runErr = r.Run(runCtx)
+
+		r.workflowTracker.Finalize(runErr == nil)
+
+		return runErr
+	}
+
+	err := r.workflowTracker.RunWithVisualization(runFnCli, updatesChannel)
+
+	if err != nil {
+		logger.Errorf("Workflow testing failed: %s", err)
+	}
+
+	// Display error logs if the workflow failed
+	if runErr != nil {
+		logger.Errorf("Workflow testing failed: %s\n", runErr)
+
+		output := strings.TrimSpace(logCaptureBuffer.String())
+
+		logger.PrintlnUnstyled(styles.MakeSection("Workflow testing run logs", output, styles.Colors.Grey))
+	}
+
+	return errors.Join(err, runErr)
+}
+
+// Checks the account type to ensure testing is enabled.
+func (r *Runner) checkAccountType(ctx context.Context) error {
+	accountType := auth.GetAccountTypeFromContext(ctx)
+
+	if accountType == nil {
+		return fmt.Errorf("Account type not found. Ensure you are logged in via the `speakeasy auth login` command or SPEAKEASY_API_KEY environment variable.")
+	}
+
+	if *accountType != shared.AccountTypeEnterprise {
+		return fmt.Errorf("Testing is not supported on the %s account tier. Contact support@speakeasy.com for more information.", *accountType)
```

Comment:
Might be worth highlighting the email with lipgloss in an emphasised style

Reply:
Do we need to be concerned about GitHub Actions output or anything with the additional formatting?